components/ledger/pkg/storage/accounts.go

```diff
@@ -0,0 +1,63 @@
+package storage
+
+import (
+	"github.com/formancehq/stack/libs/go-libs/metadata"
+)
+
+type AccountsQuery OffsetPaginatedQuery[AccountsQueryPayload]
+
+type AccountsQueryPayload struct {
+	AfterAddress    string
+	Address         string
+	Balance         string
+	BalanceOperator BalanceOperator
+	Metadata        metadata.Metadata
+}
+
+func NewAccountsQuery() AccountsQuery {
+	return AccountsQuery{
+		PageSize: QueryDefaultPageSize,
+		Order:    OrderAsc,
+		Payload: AccountsQueryPayload{
+			Metadata: metadata.Metadata{},
+		},
+	}
+}
+
+func (a AccountsQuery) WithPageSize(pageSize uint64) AccountsQuery {
+	if pageSize != 0 {
+		a.PageSize = pageSize
```

Comment:
Should we return an error when pageSize == 0 ? Maybe not here but in the caller ?